components/Badge/Badge.tsx

```diff
@@ -92,13 +77,50 @@ export const Badge = styled('span', {
       true: {},
     },
   },
-  compoundVariants: [
-    ...interactiveColorCompoundVariants,
-    ...alphaColorCompoundVariants,
-  ],
+  compoundVariants: [...interactiveColorCompoundVariants, ...alphaColorCompoundVariants],
   defaultVariants: {
     size: 'small',
     interactive: false,
     variant: 'gray',
   },
+};
+
+export const StyledSpanBadge = styled('span', BADGE_BASE_STYLES);
+
+const StyledButtonBadge = styled('button', BADGE_BASE_STYLES, {
+  '&:focus-visible': {
+    boxShadow: '0 0 0 1px $colors$focusOutline',
+  },
+  '&:hover': {
+    cursor: 'pointer',
+    '&::before': {
+      backgroundColor: '$badgeInteractiveBackground',
+      boxSizing: 'border-box',
+      content: '""',
+      position: 'absolute',
+      top: 0,
+      right: 0,
+      bottom: 0,
+      left: 0,
+      borderRadius: '5px',
+    },
+  },
 });
+
+interface BadgeProps
+  extends ComponentProps<typeof StyledButtonBadge>,
+    VariantProps<typeof StyledButtonBadge> {}
+
+export const Badge = React.forwardRef<React.ElementRef<typeof StyledButtonBadge>, BadgeProps>(
+  ({ interactive, ...props }, forwardedRef) => {
+    return (
+      <>
+        {interactive ? (
+          <StyledButtonBadge {...props} ref={forwardedRef} />
+        ) : (
+          <StyledSpanBadge {...props} ref={forwardedRef} />
+        )}
+      </>
+    );
```

Comment:
I think we can remove the fragment here:

```suggestion
    return interactive ? (
      <StyledButtonBadge {...props} ref={forwardedRef} />
    ) : (
      <StyledSpanBadge {...props} ref={forwardedRef} />
    );
```